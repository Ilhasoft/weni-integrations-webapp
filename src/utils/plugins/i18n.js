import Vue from 'vue';
import VueI18n from 'vue-i18n';

Vue.use(VueI18n);

import en from '../../locales/en.json';
import pt_br from '../../locales/pt_br.json';
import es_es from '../../locales/es_es.json';

const languages = {
  'en-us': en,
  'pt-br': pt_br,
  es: es_es,
};

const messages = Object.assign(languages);

const i18n = new VueI18n({
  locale: 'en-us',
  fallbackLocale: 'en-us',
  messages,
});

export default i18n;
