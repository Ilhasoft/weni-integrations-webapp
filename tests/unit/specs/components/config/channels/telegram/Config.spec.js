import { unnnicCallAlert as mockUnnnicCallAlert } from '@weni/unnnic-system';

jest.mock('@weni/unnnic-system', () => ({
  ...jest.requireActual('@weni/unnnic-system'),
  unnnicCallAlert: jest.fn(),
}));

import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import { singleApp } from '../../../../../../__mocks__/appMock.js';
import i18n from '@/utils/plugins/i18n';

import telegramConfig from '@/components/config/channels/telegram/Config.vue';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('TelegramConfig.vue', () => {
  let wrapper;

  let actions;
  let store;

  beforeEach(() => {
    actions = {
      updateAppConfig: jest.fn(),
      getApp: jest.fn(() => {
        return { data: singleApp };
      }),
    };

    store = new Vuex.Store({
      actions,
    });

    wrapper = shallowMount(telegramConfig, {
      localVue,
      i18n,
      store,
      propsData: {
        app: singleApp,
      },
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should be rendered properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('saveConfig()', () => {
    it('should call updateAppConfig', async () => {
      const spy = spyOn(wrapper.vm, 'updateAppConfig');
      expect(spy).not.toHaveBeenCalled();
      await wrapper.vm.saveConfig();
      expect(spy).toHaveBeenCalledTimes(1);
    });

    it('should call success alert on success', async () => {
      actions.updateAppConfig.mockImplementation(() => {
        return true;
      });

      expect(mockUnnnicCallAlert).not.toHaveBeenCalled();
      await wrapper.vm.saveConfig();
      expect(mockUnnnicCallAlert).toHaveBeenCalledTimes(1);
      expect(mockUnnnicCallAlert).toHaveBeenCalledWith(
        expect.objectContaining({
          props: {
            title: 'Success',
            text: expect.any(String),
            icon: expect.any(String),
            scheme: expect.any(String),
            position: expect.any(String),
            closeText: expect.any(String),
          },
          seconds: expect.any(Number),
        }),
      );
    });

    it('should call error alert on error', async () => {
      actions.updateAppConfig.mockImplementation(() => {
        throw new Error('error fetching');
      });

      expect(mockUnnnicCallAlert).not.toHaveBeenCalled();
      await wrapper.vm.saveConfig();
      expect(mockUnnnicCallAlert).toHaveBeenCalledTimes(1);
      expect(mockUnnnicCallAlert).toHaveBeenCalledWith(
        expect.objectContaining({
          props: {
            title: 'Error',
            text: expect.any(String),
            icon: expect.any(String),
            scheme: expect.any(String),
            position: expect.any(String),
            closeText: expect.any(String),
          },
          seconds: expect.any(Number),
        }),
      );
    });
  });

  describe('closeConfig()', () => {
    it('should emit closeModal', () => {
      expect(wrapper.emitted('closeModal')).toBeFalsy();
      wrapper.vm.closeConfig();
      expect(wrapper.emitted('closeModal')).toBeTruthy();
    });
  });
});
