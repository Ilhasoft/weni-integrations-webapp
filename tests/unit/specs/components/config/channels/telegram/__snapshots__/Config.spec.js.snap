// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TelegramConfig.vue should be rendered properly 1`] = `
<div class="app-config-telegram">
  <div class="app-config-telegram__header">
    <div class="app-config-telegram__header__title">
      <div class="app-config-telegram__header__title__icon-container"><img src="https://weni-sp-push-dev.s3.sa-east-1.amazonaws.com/svg/logo-wwc.svg" class="app-config-telegram__header__title__icon-container__icon"></div>
      <div class="app-config-telegram__header__title__name">Weni Web Chat</div>
    </div>
    <div class="app-config-telegram__header__description">
      Learn more about how to integrate Telegram
      <a href="https://docs.ilhasoft.mobi/l/pt/article/qzw2i2og1s-como-obter-o-token-de-um-bot-do-telegram#" target="_blank">
        here.
      </a>
    </div>
  </div>
  <div class="app-config-telegram__settings__content">
    <unnnic-input-stub type="normal" label="Token" tooltipsideiconright="right" size="md" mask=""></unnnic-input-stub>
  </div>
  <div class="app-config-telegram__settings__buttons">
    <unnnic-button-stub size="large" text="Cancel" type="terciary" class="app-config-telegram__settings__buttons__cancel"></unnnic-button-stub>
    <unnnic-button-stub size="large" text="Validate" type="secondary" class="app-config-telegram__settings__buttons__save"></unnnic-button-stub>
  </div>
</div>
`;
